app/src/main/java/uk/co/neuralcubes/neuralates/ControlFragment.java

```diff
@@ -254,27 +254,22 @@ public void run() {
 
     @Override
     public void updateRobots(final List<Robot> robots) {
-        this.getActivity().runOnUiThread(new Runnable() {
```

Comment:
It's not about Android's model but the SDK's. There's no documentation so I tested it empirically and it's always called on the UI thread. Usually SDKs queue callbacks in the UI thread anyway.